zone/zonedb.cpp

```diff
@@ -1516,17 +1515,35 @@ bool ZoneDatabase::SaveCharacterMemorizedSpell(uint32 character_id, uint32 spell
 	return true;
 }
 
-bool ZoneDatabase::SaveCharacterSpell(uint32 character_id, uint32 spell_id, uint32 slot_id){
-	if (spell_id > SPDAT_RECORDS){ return false; }
-	std::string query = StringFormat("REPLACE INTO `character_spells` (id, slot_id, spell_id) VALUES (%u, %u, %u)", character_id, slot_id, spell_id);
-	QueryDatabase(query);
-	return true;
+bool ZoneDatabase::SaveCharacterSpell(uint32 character_id, uint32 spell_id, uint32 slot_id)
+{
+	if (!IsValidSpell(spell_id)) {
+		return false;
+	}
+
+	auto e = CharacterSpellsRepository::NewEntity();
+
+	e.id       = character_id;
+	e.slot_id  = slot_id;
+	e.spell_id = spell_id;
+
+	const int replaced = CharacterSpellsRepository::ReplaceOne(*this, e);
```

Comment:
Inline

Reply:
In fact this could be simplified to 

```cpp
bool ZoneDatabase::SaveCharacterSpell(uint32 character_id, uint32 spell_id, uint32 slot_id)
{
	if (!IsValidSpell(spell_id)) {
		return false;
	}

	return CharacterSpellsRepository::ReplaceOne(
		*this,
		CharacterSpellsRepository::CharacterSpells{
			.id = character_id,
			.slot_id = static_cast<uint16_t>(slot_id),
			.spell_id = static_cast<uint16_t>(spell_id)
		}
	);
}
```